src/wallet/rpcwallet.cpp

```diff
@@ -4276,34 +4286,60 @@ UniValue addressmap(const JSONRPCRequest &request) {
     if (typeInt < 0 || typeInt >= AddressConversionTypeCount) {
         throwInvalidParam();
     }
-    const auto type = static_cast<AddressConversionType>(request.params[1].get_int());
+
+    CTxDestination dest = DecodeDestination(input);
+    const auto type = static_cast<AddressConversionType>(typeInt);
+
+    UniValue format(UniValue::VOBJ);
+    UniValue ret(UniValue::VOBJ);
+    ret.pushKV("input", input);
+    ret.pushKV("type", typeInt);
+
     switch (type) {
         case AddressConversionType::DVMToEVMAddress: {
-            CTxDestination dest = DecodeDestination(input);
             if (dest.index() != WitV0KeyHashType && dest.index() != PKHashType) {
                 throwInvalidParam();
             }
             CPubKey key = AddrToPubKey(pwallet, input);
             if (key.IsCompressed()) {
                 key.Decompress();
             }
-            return EncodeDestination(WitnessV16EthHash(key));
+            format.pushKV("erc55", EncodeDestination(WitnessV16EthHash(key)));
+            break;
         }
         case AddressConversionType::EVMToDVMAddress: {
-            CTxDestination dest = DecodeDestination(input);
             if (dest.index() != WitV16KeyEthHashType) {
                 throwInvalidParam();
             }
             CPubKey key = AddrToPubKey(pwallet, input);
             if (!key.IsCompressed()) {
                 key.Compress();
             }
-            return EncodeDestination(WitnessV0KeyHash(key));
+            format.pushKV("bech32", EncodeDestination(WitnessV0KeyHash(key)));
+            break;
         }
         default:
-            throw JSONRPCError(RPC_INVALID_REQUEST, "Invalid address type passed");
-            break;
+            if (dest.index() == WitV0KeyHashType || dest.index() == PKHashType) {
```

Comment:
Could you pull this logic up and  just set `AddressConversionType` when you can determine it? If not, it still does through the default case, and fails.

Reply:
If we cannot determine, let's just fail with something like, "Unsupported type or unable to determine conversion type automatically from the input".

Reply:
> Could you pull this logic up and just set AddressConversionType when you can determine it

hmm.. i don't quite understand.. why do we need to determine `AddressConversionType ` by using that logic since we ady have `type` provided from client??

Reply:
got it thanks for guidance
auto to determine type -> pass the type to switch flow will do